esmvaltool/diag_scripts/kcs/resampling.py

```diff
@@ -186,18 +188,8 @@ def get_relative_values(top1000, info, period):
 
 def determine_penalties(overlap):
     """Determine penalties dependent on the number of overlaps."""
-    return np.piecewise(
-        overlap,
-        condlist=[
-            overlap < 3,
-            overlap == 3,
-            overlap == 4,
-            overlap > 4],
-        funclist=[
-            0,
-            1,
-            5,
-            100])
+    conditions = [overlap < 3, overlap == 3, overlap == 4, overlap > 4]
+    return np.piecewise(overlap, condlist=conditions, funclist=[0, 1, 5, 100])
```

Comment:
I think this is actually slower than substituting the conditions directly into np.piecewise, because now you will first make a list of testing all the points for all the criteria. 
```suggestion
    return np.piecewise(
        overlap,
        condlist=[overlap < 3, overlap == 3, overlap == 4, overlap > 4], 
        funclist=[0, 1, 5, 100]
        )
```

Reply:
I made this change to fix the pylint length error while avoiding the style which was suggested by autopep and looks ugly. Now, it seems that this change makes the function slower. So, I changed it again to autopep suggestion.